pkg/cmd/pulumi/stack_init.go

```diff
@@ -161,5 +184,31 @@ func newStackInitCmd() *cobra.Command {
 		&stackToCopy, "copy-config-from", "", "The name of the stack to copy existing config from")
 	cmd.PersistentFlags().BoolVar(
 		&noSelect, "no-select", false, "Do not select the stack")
+	cmd.PersistentFlags().StringArrayVar(&teams, "teams", nil, "An assigment from team "+
+		"names to the permissions they should have on this stack, "+
+		"once created (Pulumi Service backend only)")
 	return cmd
 }
+
+// This function constructs a createStackOptions object if
+// valid, otherwise returning nil. Most backends expect nil
+// options, and error if options are non-nil.
+func validateCreateStackOpts(teams []string) *stackCreateOptions {
+	var opts *stackCreateOptions
+	// Remove any strings from the list that are empty or just whitespace.
+	var validatedTeams []string
+	for _, team := range teams {
+		var teamStr = strings.TrimSpace(team)
+		if len(teamStr) > 0 {
```

Comment:
optional: Since `teams` is not used again after this
you can do this without allocating a new slice.
```suggestion
	var validatedTeams = teams[:0] // re-use storage
	for _, team := range teams {
		var teamStr = strings.TrimSpace(team)
		if len(teamStr) > 0 {
```

ref: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating